compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -337,6 +337,13 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             infer_args,
         );
 
+        // Skip processing if type isn't generic.
+        // Traits have `Self` as a generic parameter, which means they will not return early here
+        // and associated type bindings will be handled regardless of the trait being generic.
```

Comment:
```suggestion
        // Skip processing if type has no generic parameters.
        // Traits always have `Self` as a generic parameter, which means they will not return early
        // here and so associated type bindings will be handled regardless of whether there are any
        // non-`Self` generic parameters.
```
I think this is slightly clearer.

Reply:
Indeed. I committed your change and squashed the commits.